core/src/test/scala/unit/kafka/admin/AdminRackAwareTest.scala

```diff
@@ -0,0 +1,198 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package kafka.admin
+
+import junit.framework.Assert._
+import kafka.common.TopicAndPartition
+import kafka.utils.{TestUtils, ZkUtils, Logging}
+import kafka.zk.ZooKeeperTestHarness
+import org.I0Itec.zkclient.ZkClient
+import org.junit.Test
+import org.scalatest.junit.JUnit3Suite
+
+import scala.collection.{mutable, Map, Seq}
+
+class AdminRackAwareTest extends ZooKeeperTestHarness with Logging with RackAwareTest {
```

Comment:
There is a test case (testGetBrokersAndRacks) testing the behavior of what broker-rack mapping should be used under different rack aware mode.